src/shared/listener-collection.js

```diff
@@ -11,24 +11,41 @@
  */
 export class ListenerCollection {
   constructor() {
-    /** @type {Listener[]} */
+    /** @type {Array<Listener|null>} */
     this._listeners = [];
   }
 
   /**
+   * Adds a listener and returns a listenerId
```

Comment:
There should be a blank line between the description of a symbol and any JSDoc tags (`@param`, `@prop` etc.).